pkg/pipeline/git_test.go

```diff
@@ -0,0 +1,349 @@
+package pipeline
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"testing"
+
+	synv1alpha1 "github.com/projectsyn/lieutenant-operator/pkg/apis/syn/v1alpha1"
+	"github.com/projectsyn/lieutenant-operator/pkg/git/manager"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestCreateOrUpdateGitRepo(t *testing.T) {
+	type args struct {
```

Comment:
There is no need to define a named type when it is only used one. You can define the variable along the data structure.

Reply:
Looks like I've missed this file when changing the tests.

FWIW: anonymous variables in golang are quite a pain, here's a very nice example: https://stackoverflow.com/a/54126129. That's the reasoning behind this definition.

Reply:
I see, but when defining the variable and the data structure within the same statement, this is not an issue. At least it never was for me.